src/logic/export/PolygonLabelsExporter.ts

```diff
@@ -0,0 +1,86 @@
+import {ExportFormatType} from "../../data/ExportFormatType";
+import {IPoint} from "../../interfaces/IPoint";
+import {VGGFileData, VGGObject, VGGPolygon, VGGRegionsData} from "../../data/VGG/IVGG";
+import {ImageData, LabelPolygon} from "../../store/editor/types";
+import {EditorSelector} from "../../store/selectors/EditorSelector";
+import {saveAs} from "file-saver";
+import moment from 'moment';
+
+export class PolygonLabelsExporter {
+    public static export(exportFormatType: ExportFormatType): void {
+        switch (exportFormatType) {
+            case ExportFormatType.VGG_JSON:
+                PolygonLabelsExporter.exportAsVGGJson();
+                break;
+            default:
+                return;
+        }
+    }
+
+    private static exportAsVGGJson(): void {
+        const imagesData: ImageData[] = EditorSelector.getImagesData();
+        const labelNames: string[] = EditorSelector.getLabelNames();
+        const outputObject: VGGObject = imagesData.reduce((data: VGGObject, image: ImageData) => {
+            const fileData: VGGFileData = PolygonLabelsExporter.mapImageDataToVGGFileData(image, labelNames);
+            if (!!fileData) {
+                data[image.fileData.name] = fileData
+            }
+            return data;
+        }, {});
+        const content: string = JSON.stringify(outputObject);
+        const projectName: string = EditorSelector.getProjectName();
+        const date: string = moment().format('YYYYMMDDhhmmss');
+        const blob = new Blob([content], {type: "text/plain;charset=utf-8"});
+        try {
+            saveAs(blob, `labels_${projectName}_${date}.json`);
```

Comment:
takie coś masz w każdym exporterze. może dałoby się to jakoś w jednym miejscu zapisać? w sensie jakiś util. żeby mieć w jednym miejscu wszystkie nazwy plików, do których exportujesz

Reply:
nie chodzi mi o samego save'a, tylko nazwy plików